### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by 86sanj**
# Peer Review Information

- Fixed /docs/index.rst: Removed train/eval details and add Dataset Explore
- /Readme: Added Dataset Explore api

# Pull Request Check List

- [ ] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -22,37 +22,34 @@ Dataset Statistics
 
 We provide a sample `notebook <https://github.com/Unity-Technologies/datasetinsights/blob/master/notebooks/SynthDet_Statistics.ipynb>`_ to help you get started with dataset statistics for the `SynthDet <https://github.com/Unity-Technologies/SynthDet>`_ project. We plan to support other sample Unity projects in the future.
```
Comment (BlairLee) [inline docs/source/index.rst]: The notebook link is invalid. Need to be updated.

**Review by BlairLee — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -27,8 +27,8 @@ datasetinsights download \
 
 ```python3
 
-from datasetinsights.io.downloader import UnitySimulationDownloader,
-GCSDatasetDownloader, HTTPDatasetDownloader
+from datasetinsights.io.downloader import (UnitySimulationDownloader,
```
Comment (adason) [inline README.md]: I will separate these three downloader into three code blocks and explain briefly what each block is trying to achieve.

File: README.md [hunk 3]
```diff
@@ -40,6 +40,37 @@ downloader = HTTPDatasetDownloader()
...
 ```
+### Dataset Explore
+You can explore the downloaded dataset [schema](https://datasetinsights.readthedocs.io/en/latest/Synthetic_Dataset_Schema.html#synthetic-dataset-schema) by using following API:
+
+[Unity Perception](https://datasetinsights.readthedocs.io/en/latest/datasetinsights.datasets.unity_perception.html#datasetinsights-datasets-unity-perception)
+
+
+```python3
+
+from datasetinsights.datasets.unity_perception import (AnnotationDefinitions,
```
Comment (adason) [inline README.md]: Let's separate metrics, captures and definitions here into separate code blocks. We can leave out egos and sensors for now as these are less used feature in the current perception package. Also, we should provide explanation in each code block to let user know what to expect.

File: docs/source/index.rst [hunk 4]
```diff
@@ -6,53 +6,78 @@
...
 
-Unity Dataset Insights is a python package for understanding synthetic datasets.
-This package enables users to analyze synthetic datasets generated using the `Perception SDK <https://github.com/Unity-Technologies/com.unity.perception>`_.
+Unity Dataset Insights is a python package for downloading, parsing and analyzing synthetic datasets generated using the Unity `Perception SDK <https://github.com/Unity-Technologies/com.unity.perception>`_.
 
 Installation
 ------------
 
-Dataset Insights maintains a pip package for easy installation. It can work in any standard Python environment using :code:`pip install datasetinsights` command. We support Python 3 (>= 3.7).
+Dataset Insights maintains a pip package for easy installation. It can work in any standard Python environment using :code:`pip install datasetinsights` command. We support Python 3 (3.7 and 3.8).
```
Comment (BlairLee) [inline docs/source/index.rst:14]: Change back to >=3.7.

File: README.md [hunk 5]
```diff
@@ -40,6 +40,37 @@ downloader = HTTPDatasetDownloader()
 downloader.download(source_uri=source_uri, output=data_root)
 
 ```
+### Dataset Explore
+You can explore the downloaded dataset [schema](https://datasetinsights.readthedocs.io/en/latest/Synthetic_Dataset_Schema.html#synthetic-dataset-schema) by using following API:
```
Comment (adason) [inline README.md]: Users don't need to download datasets in order to use this package. This package should also work for development use cases when they generated the dataset on their workstation.

File: docs/source/index.rst [hunk 6]
```diff
@@ -6,53 +6,78 @@
...
 
    datasetinsights download \
       --source-uri=<xxx> \
       --output=$HOME/data
 
-`Train <https://datasetinsights.readthedocs.io/en/latest/datasetinsights.commands.html#datasetinsights-commands-train>`_
+`API <https://datasetinsights.readthedocs.io/en/latest/datasetinsights.io.downloader.html#module-datasetinsights.io.downloader.gcs_downloader>`_
 
-.. code-block:: bash
+.. code-block:: python3
```
Comment (adason) [inline docs/source/index.rst:40]: Fix this. See comments in README

File: docs/source/index.rst [hunk 7]
```diff
@@ -6,53 +6,78 @@
...
+   downloader.download(source_uri=source_uri, output=data_root)
+
+Dataset Explore
+~~~~~~~~~~~~~~~~~~
+
+You can explore the downloaded dataset `schema <https://datasetinsights.readthedocs.io/en/latest/Synthetic_Dataset_Schema.html#synthetic-dataset-schema>`_ by using following API:
+
+`Unity Perception <https://datasetinsights.readthedocs.io/en/latest/datasetinsights.datasets.unity_perception.html#datasetinsights-datasets-unity-perception>`_
+
+.. code-block:: python3
```
Comment (adason) [inline docs/source/index.rst:81]: Fix this. See comments in README

**Review by adason — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -25,20 +25,54 @@ datasetinsights download \
...
-```python3
-
-from datasetinsights.io.downloader import UnitySimulationDownloader,
-GCSDatasetDownloader, HTTPDatasetDownloader
+UnitySimulationDownloader downloads a dataset from Unity Simulation.
 
+```python3
+from datasetinsights.io.downloader import UnitySimulationDownloader
 downloader = UnitySimulationDownloader(access_token=access_token)
 downloader.download(source_uri=source_uri, output=data_root)
```
Comment (adason) [inline README.md]: I will suggest adding concrete variable 

```python3
from datasetinsights.io.downloader import UnitySimulationDownloader

source_uri = "usim://<some example here>"
dest = "~/data"
access_token = "XXX" 

downloader = UnitySimulationDownloader(access_token=access_token)
downloader.download(source_uri=source_uri, output=dest)
```

File: README.md [hunk 9]
```diff
@@ -25,20 +25,54 @@ datasetinsights download \
...
-
+```
+GCSDatasetDownloader downloads a dataset from GCS location.
+```python3
+from datasetinsights.io.downloader import GCSDatasetDownloader
 downloader = GCSDatasetDownloader()
 downloader.download(source_uri=source_uri, output=data_root)
```
Comment (adason) [inline README.md]: Use concrete example

File: README.md [hunk 10]
```diff
@@ -25,20 +25,54 @@ datasetinsights download \
...
-
+```
+HTTPDatasetDownloader downloads a dataset from any HTTP(S) location.
+```python3
+from datasetinsights.io.downloader import HTTPDatasetDownloader
 downloader = HTTPDatasetDownloader()
 downloader.download(source_uri=source_uri, output=data_root)
```
Comment (adason) [inline README.md]: use concrete example.

File: README.md [hunk 11]
```diff
@@ -25,20 +25,54 @@ datasetinsights download \
...
+
+AnnotationDefinitions and MetricDefinitions loads synthetic dataset definition tables and return a dictionary containing the definitions.
+
+```python3
+from datasetinsights.datasets.unity_perception import AnnotationDefinitions,
+MetricDefinitions
+annotation_def = AnnotationDefinitions(data_root="/data", version="my_schema_version")
+definition_dict = annotation_def.get_definition(def_id="my_definition_id")
+
+metric_def = MetricDefinitions(data_root="/data", version="my_schema_version")
```
Comment (adason) [inline README.md]: The path here should be a follow after the dataset was downloaded. I suggest using a consistent path as in the download example.

**Review by adason — CHANGES_REQUESTED** [review]

File: README.md [hunk 12]
```diff
@@ -4,7 +4,7 @@ Unity Dataset Insights is a python package for downloading, parsing and analyzin
 
 ## Installation
 
-Dataset Insights maintains a pip package for easy installation. It can work in any standard Python environment using `pip install datasetinsights` command. We support Python 3 (3.7 and 3.8).
+Dataset Insights maintains a pip package for easy installation. It can work in any standard Python environment using `pip install datasetinsights` command. We support Python 3 (>= 3.7).
```
Comment (adason) [inline README.md]: Why is this changed? Are we supporting python 3.9 now?

**Review by BlairLee — APPROVED** [review]

**Review by adason — APPROVED** [review]

**Pull request merged**